### PR TITLE
Compatible with MySQL 5.7 and MariaDB 10.2.6+

### PR DESCRIPTION
**Pull request by emqplus**
Fix issue https://github.com/emqtt/emq-auth-mysql/issues/62
Fix issue https://github.com/mysql-otp/mysql-otp/issues/72

### CODE REVIEW

File: src/mysql_protocol.erl [hunk 1]
```diff
@@ -72,7 +72,8 @@ handshake_finish_or_switch_auth(Handshake, Password, SockModule, Socket,
     {ok, ConfirmPacket, SeqNum1} = recv_packet(SockModule, Socket, SeqNum0),
     case parse_handshake_confirm(ConfirmPacket) of
         #ok{status = OkStatus} ->
-            OkStatus = Handshake#handshake.status,
+            true = (OkStatus == Handshake#handshake.status)
+                    orelse (OkStatus == (16#4000 bor Handshake#handshake.status)),
```
Comment (zuiderkwast) [inline src/mysql_protocol.erl]: 1. Please add a comment before these two lines. Suggestion:

```erlang
            %% check status, ignoring bit 16#4000, SERVER_SESSION_STATE_CHANGED.
```

2. Why not
```erlang
            true = OkStatus band bnot 16#4000 == Handshake#handshake.status,
```

I think it is more ideomatic to my understanding of it: _match, ignoring bit 16#4000" in OkStatus_

Comment (emqplus) [reply src/mysql_protocol.erl]: @zuiderkwast Cannot match a record field in Erlang. I updated the PR.

```
ERLC   mysql_protocol.erl
src/mysql_protocol.erl:77: illegal pattern
make[1]: *** [ebin/mysql.app] Error 1
make: *** [app] Error 2
```

**Comment by zuiderkwast** [comment]
Thanks!

**Comment by emqplus** [comment]
@zuiderkwast Happy new year. Could we tag 1.3.1 for the PR? We are planning to release EMQ 2.3.3 which depends on the latest mysql-otp.

**Comment by zuiderkwast** [comment]
Yes.... will do

**Pull request merged**